geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelQueueSetPossibleDuplicateMessage.java

```diff
@@ -0,0 +1,166 @@
+
+  private static final Logger logger = LogService.getLogger();
+
+  private int reason;
+  private Map regionToDispatchedKeysMap;
+
+  public static final int UNSUCCESSFULLY_DISPATCHED = 0;
+  public static final int RESET_BATCH = 1;
+  public static final int LOAD_FROM_TEMP_QUEUE = 2;
+  public static final int STOPPED_GATEWAY_SENDER = 3;
+
+
+  public ParallelQueueSetPossibleDuplicateMessage() {}
+
+  public ParallelQueueSetPossibleDuplicateMessage(int reason, Map rgnToDispatchedKeysMap) {
+    this.reason = reason;
+    this.regionToDispatchedKeysMap = rgnToDispatchedKeysMap;
+  }
+
+  @Override
+  public int getDSFID() {
+    return PARALLEL_QUEUE_SET_POSSIBLE_DUPLICATE_MESSAGE;
+  }
+
+  @Override
+  public String toString() {
+    String cname = getShortClassName();
+    final StringBuilder sb = new StringBuilder(cname);
+    sb.append("reason=" + reason);
+    sb.append(" regionToDispatchedKeysMap=" + regionToDispatchedKeysMap);
+    sb.append(" sender=").append(getSender());
+    return sb.toString();
+  }
+
+  @Override
+  protected void process(ClusterDistributionManager dm) {
+    final boolean isDebugEnabled = logger.isDebugEnabled();
+    final InternalCache cache = dm.getCache();
+
+    if (cache != null) {
+      final InitializationLevel oldLevel =
+          LocalRegion.setThreadInitLevelRequirement(BEFORE_INITIAL_IMAGE);
+      try {
+        for (Object name : regionToDispatchedKeysMap.keySet()) {
+          final String regionName = (String) name;
+          final PartitionedRegion region = (PartitionedRegion) cache.getRegion(regionName);
+          if (region == null) {
+            continue;
+          } else {
+            AbstractGatewaySender abstractSender = region.getParallelGatewaySender();
+            // Find the map: bucketId to dispatchedKeys
+            // Find the bucket
+            // Destroy the keys
+            Map bucketIdToDispatchedKeys = (Map) this.regionToDispatchedKeysMap.get(regionName);
+            for (Object bId : bucketIdToDispatchedKeys.keySet()) {
+              final String bucketFullPath =
+                  SEPARATOR + PartitionedRegionHelper.PR_ROOT_REGION_NAME + SEPARATOR
+                      + region.getBucketName((Integer) bId);
+              BucketRegionQueue brq =
+                  (BucketRegionQueue) cache.getInternalRegionByPath(bucketFullPath);
+
+              if (brq != null) {
+                if (reason == STOPPED_GATEWAY_SENDER) {
+                  brq.setReceivedGatewaySenderStoppedMessage(true);
+                }
+              }
+
+              if (isDebugEnabled) {
+                logger.debug(
+                    "ParallelQueueSetPossibleDuplicateMessage : The bucket in the cache is bucketRegionName : {} bucket: {}",
+                    bucketFullPath, brq);
+              }
+
+              List dispatchedKeys = (List) bucketIdToDispatchedKeys.get((Integer) bId);
+              if (dispatchedKeys != null && !dispatchedKeys.isEmpty()) {
```

Comment:
This seems to suggest the insertion of `emptyList()` into the map previously commented was not necessary.
.